src/Microsoft.ML.Vision/ImageClassificationTrainer.cs

```diff
@@ -1123,11 +1133,20 @@ private int GetNumSamples(string path)
 
             trainSaver.save(_session, _checkpointPath);
             UpdateTransferLearningModelOnDisk(_classCount);
+            TryCleanupTemporaryWorkspace();
+        }
+
+        private void TryCleanupTemporaryWorkspace()
+        {
+            if (_cleanupWorkspace && Directory.Exists(_options.WorkspacePath))
+            {
+                Directory.Delete(_options.WorkspacePath, true);
```

Comment:
Do you want a try-catch here? Or if this throws an exception, do you want it to kill the training process?